rcl/test/rcl/common.hpp

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__COMMON_HPP_
+#define RCL__COMMON_HPP_
+
+#include "rcl/client.h"
+#include "rcl/service.h"
+#include "rcl/rcl.h"
+
+bool
+wait_for_server_to_be_available(
+  rcl_node_t * node,
+  rcl_client_t * client,
+  size_t max_tries,
+  int64_t period_ms);
+
+bool
+wait_for_client_to_be_ready(
+  rcl_client_t * client,
+  rcl_context_t * context,
+  size_t max_tries,
+  int64_t period_ms);
+
+bool
+wait_for_service_to_be_ready(
+  rcl_service_t * service,
+  rcl_context_t * context,
+  size_t max_tries,
+  int64_t period_ms);
+
+bool
+wait_for_established_subscription(
+  const rcl_publisher_t * publisher,
+  size_t max_tries,
+  int64_t period_ms);
+
+bool
+wait_for_subscription_to_be_ready(
```

Comment:
I think this one (and maybe others) are only used in one file right now, so moving it to `common` is not strictly needed.
It's still something that might be useful for a new test, so I think that it's fine to move it here.